apps/mobile-wallet/src/hooks/useBiometrics.ts

```diff
@@ -163,30 +164,34 @@ export const useBiometricsAuthGuard = () => {
   const biometricsRequiredForTransactions = useAppSelector((s) => s.settings.requireAuth)
   const { triggerBiometricsPrompt } = useBiometricPrompt()
 
-  const triggerBiometricsAuthGuard = async ({
-    successCallback,
-    failureCallback,
-    onPromptDisplayed,
-    settingsToCheck
-  }: {
-    settingsToCheck: 'appAccess' | 'transactions' | 'appAccessOrTransactions'
-    successCallback: () => void
-    failureCallback?: () => void
-    onPromptDisplayed?: () => void
-  }) => {
-    const isBiometricsAuthRequired = {
-      appAccess: biometricsRequiredForAppAccess,
-      transactions: biometricsRequiredForTransactions,
-      appAccessOrTransactions: biometricsRequiredForAppAccess || biometricsRequiredForTransactions
-    }[settingsToCheck]
-
-    if (isBiometricsAuthRequired) {
-      onPromptDisplayed && onPromptDisplayed()
-      await triggerBiometricsPrompt({ successCallback, failureCallback })
-    } else {
-      successCallback()
-    }
-  }
+  const triggerBiometricsAuthGuard = useCallback(
+    async ({
+      successCallback,
+      failureCallback,
+      onPromptDisplayed,
+      settingsToCheck
+    }: {
+      settingsToCheck: 'appAccess' | 'transactions' | 'appAccessOrTransactions'
+      successCallback: () => void
+      failureCallback?: () => void
+      onPromptDisplayed?: () => void
+    }) => {
+      const isBiometricsAuthRequired = {
+        appAccess: biometricsRequiredForAppAccess,
+        transactions: biometricsRequiredForTransactions,
+        appAccessOrTransactions: biometricsRequiredForAppAccess || biometricsRequiredForTransactions
+      }[settingsToCheck]
+
+      if (isBiometricsAuthRequired) {
+        onPromptDisplayed && onPromptDisplayed()
+        await triggerBiometricsPrompt({ successCallback, failureCallback })
+      } else {
+        successCallback()
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [biometricsRequiredForAppAccess, biometricsRequiredForTransactions]
+  )
```

Comment:
Simply wrapping in a `useCallback` to make sure that a new function is created when the Redux state changes.